File: encrypt.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <random>
#include <ctime>
#include <cstdint>
#include "encrypt.h"
#include "ssssen.h"


using namespace std;


// Генерация случайного ключа
int generate_key() {
    random_device rd;
    mt19937 gen(rd());
    uniform_int_distribution<> dist(1, 255);
    return dist(gen);
}


int encrypt(string inFile) {
    string outFile = "encrypted.txt";

    ifstream fin(inFile, ios::binary);
    if (!fin) {
        cerr << "Не удалось открыть файл " << inFile << endl;
        return 1;
    }
    vector<char> data((istreambuf_iterator<char>(fin)), istreambuf_iterator<char>());
    fin.close();

    int key = generate_key();
    cout << "Случайный ключ: " << key << endl;

    // Разбиваем ключ
    int n = 5, k = 3;
    auto shares = shamir_split(key, n, k);

    cout << "Секрет разделен на части (для n=" << n << ", k=" << k << "):" << endl;
    for (auto &sh : shares) {
        cout << "  (" << sh.first << ", " << sh.second << ")" << endl;
    }

    // Шифруем XOR
    for (char &c : data) {
        c ^= key;
    }

    ofstream fout(outFile, ios::binary);
    fout.write(data.data(), data.size());
    fout.close();

    cout << "Файл зашифрован и сохранен в " << outFile << endl;
    return 0;
}

File: Encrypt.java
import java.io.*;
import java.util.*;

public class Encrypt {
    public static void run() {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter file path to encrypt: ");
        String path = sc.nextLine();

        try {
            byte[] data = java.nio.file.Files.readAllBytes(new File(path).toPath());
            Random rand = new Random();
            int key = 1 + rand.nextInt(255); // [1..255]

            // XOR-шифрование
            byte[] encrypted = new byte[data.length];
            for (int i = 0; i < data.length; i++) {
                encrypted[i] = (byte)(data[i] ^ key);
            }

            File out = new File("encrypted.txt");
            try (FileOutputStream fos = new FileOutputStream(out)) {
                fos.write(encrypted);
            }
            System.out.println("File encrypted: " + out.getAbsolutePath());

            // Разделение ключа Шамира (5 частей, порог = 3)
            List<int[]> shares = SSSSen.shamirSplit(key, 5, 3);
            System.out.println("Key shares:");
            for (int[] s : shares) {
                System.out.println(s[0] + " " + s[1]);
            }

        } catch (IOException e) {
            System.out.println("File error: " + e.getMessage());
        }
    }
}
